libs/math/fraction.go

```diff
@@ -38,5 +37,12 @@ func ParseFraction(f string) (Fraction, error) {
 	if err != nil {
 		return Fraction{}, fmt.Errorf("incorrect formatting, err: %w", err)
 	}
-	return Fraction{Numerator: numerator, Denominator: denominator}, nil
+	if denominator == 0 {
+		return Fraction{}, errors.New("denominator can't be 0")
+	}
+	if numerator < 0 || denominator < 0 {
```

Comment:
we can use strconv.ParseUint functions instead

Reply:
should we also check they are not greater than `math.MaxInt64` to prevent possible overflow?

Reply:
I think, looking at the code, that strconv.ParseUint should also ensure that it is less than MaxInt64

Reply:
uint64 and int64 have different max values, so no

Reply:
ah you mean in the case that we convert it. Yeah that makes sense